sources/academy/webscraping/anti_scraping/mitigation/cloudflare_challenge.md

```diff
@@ -28,4 +28,8 @@ const crawler = new PlaywrightCrawler({
 
 It's important to note that by removing default blocked status code handling, you should also add custom session retire logic on blocked pages to reduce retries. Additionally, you should add waiting logic to start the automation logic only after the Cloudflare challenge is solved and the page is redirected. This can be accomplished by waiting for a common selector that is available on all pages, such as a header logo.
 
+In some cases, the browser may not pass the check and you may be presented with a captcha, indicating that your IP address has been graylisted. To fix this, you'll need to solve the captcha to improve your IP address's reputation. There are various captcha-solving services available, such as [AntiCaptcha](https://anti-captcha.com/) or [AnyCaptcha](https://anycaptcha.com/), that you can use for this purpose. For more info check the section about [Captchas](../techniques/captchas.md).
```

Comment:
Isn't it just better to retry? We never solve captchas for blocking

Reply:
I will add this. In our case, it does not make much sense because we have a large pool of proxies, but if you have 10-20, keeping them alive and healthy is a better idea.

Reply:
Ok, we can leave it like that